### PR TITLE
anomaly localization integration step 3

### PR DESCRIPTION
**Pull request by wnbts**
Signed-off-by: lai <57818076+wnbts@users.noreply.github.com>

### Description
the changes are for step [3](https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md#step-3-add-output-class-1) in anomaly localization integration
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java [hunk 1]
```diff
@@ -110,15 +185,130 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
...
+                        entity.setContributionValue(parser.doubleValue());
+                        break;
+                    case Output.Entity.FIELD_BASE_VALUE:
+                        parser.nextToken();
+                        entity.setBaseValue(parser.doubleValue());
+                        break;
+                    case Output.Entity.FIELD_NEW_VALUE:
+                        parser.nextToken();
+                        entity.setNewValue(parser.doubleValue());
+                        break;
```
Comment (ylwu-amzn) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java:306]: Skip children for default case? Same for other places
```
                default:
                    parser.skipChildren();
                    break;
```

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java [hunk 2]
```diff
@@ -12,47 +12,89 @@
...
 
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
 /**
  * Output of localized results.
  */
 @Data
-public class Output implements ToXContent {
+@NoArgsConstructor
+public class Output implements org.opensearch.ml.common.parameter.Output {
```
[the remark this answers is not in the document]
Comment (wnbts) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java]: done

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java (hunk 1 above)
Comment (wnbts) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Output.java]: done

**Review by ylwu-amzn — DISMISSED** [review]
LGTM. Thanks for the change!

**Review by jackiehanyang — DISMISSED** [review]

**Pull request merged**